core/templates/components/main-nav/main-nav.twig

```diff
@@ -0,0 +1,44 @@
+{%- if markup is not empty %}
+  {# If custom markup is provided, emit it as is #}
+  {{ markup }}
+{% else -%}
+  {# If no custom markup, generate markup based on structured data #}
+<nav {{- attributes }} class="su-main-nav {{ modifier_class }}" aria-label="Main">
+  <button class="su-main-nav__toggle" aria-expanded="false">{{ toggle_text }}</button>
+  {% if lv1 is not empty -%}
+    <ul class="su-main-nav__menu-lv1">
+      {% for item in lv1 %}
+        {% set is_parent = item.lv2 is empty ? false : true %}
+
+        <li class="{% if is_parent %}su-main-nav__item--parent{% endif %} {% if item.current %}su-main-nav__item--current{% endif %}">{% spaceless %}
```

Comment:
Totally happy to wrap the whole UL in a `{% spaceless %}` wrapper. I am very ok with condensed output and human-readable source.